### ISSUE TITLE
Masked date with pattern and Moment.js cannot keep the mask placeholder while editing date

### ISSUE THREAD

**Issue by kpluta** [uNmAnNeR/imaskjs#436]
Hi!

I have a problem with the usage of date masking with Moment.js library, which is reproducable on the documentation page under Masked date section (https://imask.js.org/guide.html#masked-date) - demo of usage with moment. 

**Issue description:** 
Type a date, for example `1999/11/18 12:34`, then put cursor after last number of the year (let imagine that you want to change year from 1999 to 1998), then press Backspace key to remove last year number.  
Instead of clearing last year number and show placeholder, it clears a number from Date and push all other numbers to deleted position:  
![image](https://user-images.githubusercontent.com/71706792/106901337-ea5f4700-66f7-11eb-8934-90c4bac2fd18.png)
  
I reproduced same case with different date format here: https://jsfiddle.net/m8t2h1bj/7/ 
Problem is the same here, because it can hold the mask placeholder only if you edit first character.

Is there a way to keep the mask placeholder while editing the dates?

**Comment by uNmAnNeR** [comment]
hi, fiddle works for me, can not reproduce on your example. if you want to prevent moving chars between blocks, try using backquote (date mask is actually pattern mask). Feel free to reopen if you give/need more details

**Comment by luchn49** [comment]
I have the same issue. I try on the documentation.
![Screen Shot 2021-04-14 at 17 42 55](https://user-images.githubusercontent.com/57866540/114698437-4cb47380-9d49-11eb-958c-6f8ca994a9eb.png)
![Screen Shot 2021-04-14 at 17 42 59](https://user-images.githubusercontent.com/57866540/114698445-4e7e3700-9d49-11eb-9197-fd297e5bc72f.png)

@uNmAnNeR

**Comment by clukhei** [comment]
I faced the same issue as well.  I saw that @uNmAnNeR suggested using backtick, but what i found is that backtick helps prevent dates from moving backwards only for the second digit 

say the case i've filled `16/02/2024` , I specifically move cursor to change first digit of `dd` , it becomes `6d\02\2024`, but i would expect `d6/02/2024`. 

when i delete `6` from the `16/02/2024`, it works fine `1d/02/2024`. 

See recording on the example shown in imask docs page.  

I used this pattern:  `"d{/}`m{/}`Y"`

any suggestion @uNmAnNeR on how to prevent chars from moving for first digit deletion? 

https://github.com/uNmAnNeR/imaskjs/assets/55618945/9722ef76-1912-4d8d-9588-94f7061bbcc2
